float32[4] goal
---
bool success
string message
